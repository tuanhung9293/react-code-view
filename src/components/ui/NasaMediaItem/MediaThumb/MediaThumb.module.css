.wrapper {
    position: relative;
    background: black;
}

.thumb {
    width: 100%;
    min-height: 145px;
}

.playButton {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}