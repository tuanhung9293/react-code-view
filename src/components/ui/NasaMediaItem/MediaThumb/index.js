import React from 'react'
import PlayButton from 'components/ui/buttons/PlayButton'
import styles from './MediaThumb.module.css'

class MediaThumb extends React.Component {
  componentDidMount = () => {
    const {source, onFetchCollectionUrl} = this.props
    if (source === 'medias') onFetchCollectionUrl()
  }

  componentDidUpdate = (nextProps) => {
    if (nextProps.itemData.href !== this.props.itemData.href) {
      const {source, onFetchCollectionUrl} = this.props
      if (source === 'medias') onFetchCollectionUrl()
    }
  }

  render() {
    const {itemData, source} = this.props

    return (
      <div className={styles.wrapper}>
        <img src={itemData.thumbUrl} className={styles.thumb} />
        <div className={styles.playButton}>
          <PlayButton to={`/play-media/${source}/${itemData.nasaId}`} />
        </div>
      </div>
    )
  }
}

export default MediaThumb