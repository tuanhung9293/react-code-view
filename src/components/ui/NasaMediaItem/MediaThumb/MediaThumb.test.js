import React from 'react'
import PropTypes from 'prop-types'
import createRouterContext from 'react-router-test-context'
import {shallow, mount} from 'enzyme'
import MediaThumb from './'

const itemData = {
  mediaType: 'image',
  nasaId: 'nasaId',
  thumbUrl: 'thumbUrl',
  href: 'href-mock-1'
}

it('render MediaThumb', () => {
  expect(shallow(
    <MediaThumb
      itemData={itemData} source="medias"
      onFetchCollectionUrl={jest.fn(() => {})}/>
  )).toMatchSnapshot()
})

it('onFetchCollectionUrl on componentDidMount', () => {
  const context = createRouterContext()
  const childContextTypes = {
    router: PropTypes.object
  }
  
  const wrapper = mount(
    <MediaThumb
      itemData={itemData}
      source="medias"
      onFetchCollectionUrl={jest.fn(() => {})} />,
      {context, childContextTypes}
  )

  expect(wrapper.props().onFetchCollectionUrl).toBeCalled()
})

it('onFetchCollectionUrl on componentDidUpdate if change props itemData.href', () => {
  const context = createRouterContext()
  const childContextTypes = {
    router: PropTypes.object
  }
  
  const wrapper = mount(
    <MediaThumb
      itemData={itemData}
      source="medias"
      onFetchCollectionUrl={jest.fn(() => {})} />,
      {context, childContextTypes}
  )

  wrapper.setProps({itemData: {
    mediaType: 'image',
    nasaId: 'nasaId',
    thumbUrl: 'thumbUrl',
    href: 'href-mock-2'
  }})

  expect(wrapper.props().onFetchCollectionUrl).toBeCalled()
})
