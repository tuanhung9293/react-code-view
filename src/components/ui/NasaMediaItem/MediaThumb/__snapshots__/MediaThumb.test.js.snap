// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MediaThumb 1`] = `
<div
  className="wrapper"
>
  <img
    className="thumb"
    src="thumbUrl"
  />
  <div
    className="playButton"
  >
    <PlayButton
      to="/play-media/medias/nasaId"
    />
  </div>
</div>
`;
